objgen.py

```diff
@@ -0,0 +1,76 @@
+import json
+import os
+from collections import defaultdict
+from ipaddress import IPv4Network, IPv6Network
+from typing import List
+
+
+class AnnouncementGenerator:
```

Comment:
I am not sure I follow what this class does.  I think what we would want is one class for each entity people can reason about, and then create methods in each class to generate instances of that class.  Classes that make sense to me:

* PrefixAnnouncement:  Specifies how one prefix should be announced.  Contains parameters such as the origin ASN, the number of prepends, and the upstream muxes.  We could also consider adding poisoning as a parameter, even if it will not be allowed in the REST API.

* ExperimentRound:  Just a `list[PrefixAnnouncement]`

* Experiment:  Just a `list[ExperimentRound]`

Reply:
I think we could create `dataclasses` for each of these, as they mostly hold information, and then provide verification- and initialization-related functions in the `dataclass`.

Reply:
I'm wondering if we need to create a data class for the Experiment Round, since a round is a list of announcements. My idea was to create a method that adds a round, passing a list of announcements as arguments.